src/main/java/httpserver/HttpPresenter.java

```diff
@@ -4,12 +4,20 @@
 
     public String present(HttpResponse response) {
         String responseLine = response.getResponseLine() + System.lineSeparator();
+        return responseLine + headers(response) + body(response);
+    }
+
+    private String body(HttpResponse response) {
         String body = response.getBody();
-        return responseLine + headers(response) + body;
+        if(body == null) {
```

Comment:
👍 
Optional's are awesome. #provenFact

Reply:
BTW, `Optional` is similar to the idea of a result type, except instead of success or error, it represents a value or no value. This makes it a nice way to deal with nullable values without explicit if statements like this one.

```
Optional.ofNullable(body).orElse("")
```

Reply:
Removed entirely